.github/actions/setup-project/action.yml

```diff
@@ -0,0 +1,75 @@
+name: Setup Novu Monorepo
+
+description: Sets up the whole monorepo and install dependencies
+
+inputs:
+  cypress_version:
+    description: 'The version of cypress to install'
+    required: false
+    default: ''
+outputs:
+  cypress_cache_hit:
+    description: 'Did cypress used binary cache'
+    value: ${{ inputs.cypress_version != '' && steps.cache-cypress-binary-version.outputs.cache-hit || steps.cache-cypress-binary.outputs.cache-hit}}
+
+runs:
+  using: composite
+  steps:
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+    - name: ⚙️ Setup kernel for react native, increase watchers
+      shell: bash
+      run: echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p
+
+    - uses: actions/setup-node@v2
+      name: ⚙️ Setup Node Version
+      with:
+        node-version: '16.15.1'
+
+    - name: 💵 Start Redis
+      uses: supercharge/redis-github-action@1.2.0
+
+    - name: 📚 Start MongoDB
+      uses: supercharge/mongodb-github-action@1.3.0
+      with:
+        mongodb-version: 4.2.8
```

Comment:
I'm not sure if we would like to do this for all the workflows, ex. the `dev-deploy-embed` doesn't need it.
I feel like both should be optional.

Reply:
I aggree, in the next phase we will optimize those areas. Will create a task for  this. Wasn't part of the scope for this change